### DISCUSSION TITLE
A field formatter that skips displaying a certain field

### DISCUSSION THREAD

**Discussion by alcroito** [tokio-rs/tracing#2863] [Q&A]
Hi,

I create tracing events with a boolean marker field like
```
# `e` is an `eyre` `Report`.
error!(has_emoji = true, some_other_field = "value", "{e:#}");
```

I then implement `tracing_subscriber::fmt::FormatEvent` for my custom event formatter, something like:
```rust

impl<S, N> tracing_subscriber::fmt::FormatEvent<S, N> for EmojiFmt
where
    S: tracing::subscriber::Subscriber + for<'a> tracing_subscriber::registry::LookupSpan<'a>,
    N: for<'a> tracing_subscriber::fmt::FormatFields<'a> + 'static,
{
    fn format_event(
        &self,
        ctx: &tracing_subscriber::fmt::FmtContext<'_, S, N>,
        mut writer: tracing_subscriber::fmt::format::Writer<'_>,
        event: &tracing::Event<'_>,
    ) -> Result<(), std::fmt::Error> {
        // Detect whether the message already has an emoji, so we don't add another one.
        let mut visitor = HasEmojiVisitor { has_emoji: false };
        event.record(&mut visitor);

        // Get the emoji for the log level, unless the message already has an emoji.
        let maybe_emoiji_message = {
            if visitor.has_emoji {
                "".to_owned()
            } else {
                let level_emoji = get_emoji_for_log_level(event.metadata().level());
                format!("{} ", level_emoji)
            }
        };
        write!(writer, "{}", maybe_emoiji_message)
            .expect("Invalid message write in tracing's format_event");

        // TODO: Get rid of the has_emoji field at the end after the message somehow
        use tracing_subscriber::fmt::FormatFields;
        ctx.format_fields(writer.by_ref(), event)?;
        writeln!(writer)
    }
}
```

The problem is that when I display the event with `ctx.format_fields(writer.by_ref(), event)?;` at the bottom, it also prints the `has_emoji = true` part, which i'd like to skip.

I scoured the docs, and found https://docs.rs/tracing-subscriber/0.3.18/tracing_subscriber/field/index.html but none of the utilities there allows skipping a field.

How could I got about implementing this, while keeping the default formatting?

Is copying and adapting the whole of 
https://docs.rs/tracing-subscriber/0.3.18/tracing_subscriber/fmt/format/struct.DefaultFields.html
and
https://docs.rs/tracing-subscriber/0.3.18/tracing_subscriber/fmt/format/struct.DefaultVisitor.html
the only way of doing this?

Or perhaps there's a different better way?
Thanks!

**Comment by kummatty** [comment]
Did you find a solution for this?
